eng/AnalyzersForTemplateEngine.props

```diff
@@ -0,0 +1,10 @@
+<Project>
+  <PropertyGroup>
+    <!-- Disable analyzers in sourcebuild -->
+    <EnableAnalyzers Condition="'$(DotNetBuildFromSource)' == 'true'">false</EnableAnalyzers>
+  </PropertyGroup>
+  <ItemGroup Condition="'$(EnableAnalyzers)' == 'true'">
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.CodeStyle" Version="$(MicrosoftCodeAnalysisCSharpCodeStylePackageVersion)" PrivateAssets="all" />
+    <PackageReference Include="StyleCop.Analyzers" Version="1.2.0-beta.435" PrivateAssets="all" />
```

Comment:
`StyleCop` can be a trouble. Let's enable it only for our projects: `TemplateEngine.Cli`, `dotnet-new3`, our tests projects.

```
<ItemGroup Condition="$(EnableAnalyzers)' == 'true' And '$(EnableStyleCopAnalyzer)' == 'true'">
    <PackageReference Include="StyleCop.Analyzers" Version="1.2.0-beta.435" PrivateAssets="all" />
</ItemGroup>
```

and in respective projects
```
<EnableStyleCopAnalyzer">true</EnableStyleCopAnalyzer>
```